### PR TITLE
Request a review

### PR DESCRIPTION
**Pull request by maelfosso**
Please could you review my code ?

### CODE REVIEW

File: bubble-sort.rb [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+        array[j] = array[j + 1]
+        array[j + 1] = tmp
+      end
+    end
+  end
+
+  return array
+end
+
+def bubble_sort_by(array)
```
Comment (inhaq) [inline bubble-sort.rb:16]: this method isn't working without block, kindly check for whether the block is given or not. Also in the project, it's required to use spaceship operator for comparison.

**Review by inhaq — CHANGES_REQUESTED** [review]
Welcome to Ruby world. You've done a good job on your first try. Kindly look into the comment and fix it accordingly. Once fixed, kindly submit for review.

File: bubble-sort.rb (hunk 1 above)
Comment (maelfosso) [reply bubble-sort.rb:16]: Please about *spaceship*, it's instead said that 

> The block’s return will be similar to the spaceship operator you learned about before: If the result of the block execution is negative, the element on the left is “smaller” than the element on the right. 0 means both elements are equal. A positive result means the left element is greater

It said `Similar` not that with have to use it

Comment (inhaq) [reply bubble-sort.rb:16]: Sorry, my bad. Thanks for bringing that up 😊
Although, you need to fix it for a block.

File: bubble-sort.rb [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+  return nil if !block_given?
```
Comment (inhaq) [inline bubble-sort.rb]: It should not return `nil` when no block is given. `bubble_sort_by` needs to work with and without a block. BTW good work with `spaceship` operator. It's always great to learn newer syntax.

Comment (maelfosso) [reply bubble-sort.rb]: OK. Thanks

File: bubble-sort.rb [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+      end
+    end
+  end
+
+  return array
+end
+
+def bubble_sort_by(array)
+
+  return bubble_sort(array) if !block_given?
```
Comment (inhaq) [inline bubble-sort.rb:18]: Good one 😊you can call for the `bubble_sort` when no block is given.

**Review by inhaq — APPROVED** [review]
👍🏻for the project. Go ahead and submit a project completion form. 
Good luck with the next project.

**Pull request merged**